### PR TITLE
fix(tables): correct table styles [1.x]

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-1255 (Part 1 of 2)

This PR addresses the styling updates needed to correct column width calculations. There are some JS updates to correct alignment for paginated data that resizes columns as the header is floating.

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM

### CODE REVIEW

File: src/components/rxApp/common.less [hunk 1]
```diff
@@ -150,6 +190,18 @@ table {
...
+    tbody {
+      border: 0;
+    }
+
+    tfoot {
+      border: 0;
+    }
+
     th:first-child {
       border-left-width: 0;
```
Comment (CITguy) [inline src/components/rxApp/common.less]: Note to self: remove this style definition. It's irrelevant.

**Comment by CITguy** [comment]
## Borders
Borders had to be restyled to work with corrected styling. (It's not possible to do header-only border styling with correction.)

### Before
screen shot 2017-01-30 at 2 44 07 pm

* slightly darker border on left, top, and right of header (cannot accomplish across supported browsers)

### After
screen shot 2017-01-30 at 2 43 43 pm

* Table perimeter has consistent border color (header background is same color as table border).

**Comment by CITguy** [comment]
## Nested Tables

* Color contrast of chevron arrows was not ADA Compliant (1.2)

### Before
screen shot 2017-01-30 at 2 50 13 pm

* expanded chevron color contrast is only 1.2 (not ADA compliant)

### After
screen shot 2017-01-30 at 2 50 23 pm

* expanded chevron color is left alone as the background changes (results in a 3.7 contrast ratio)

**Comment by CITguy** [comment]
# rxBulkSelect

## BEFORE

### Chrome
chrome-before

* double border (table width changes) on right side of floating header
* shifting column widths when floating

### Firefox
ff-before

* double border (table width changes) on left side of floating header
* shifting column widths when floating

## AFTER
* Fixed double border issue when floating.

### Chrome
jan-30-2017 14-54-01

* Small UI bug with webkit when using `rxFloatingHeader` with `rxBulkSelect` **only when the bulk select header is visible**
  * Table columns seem to shift a bit when floating and something is selected.
  * This is believed to be a rendering bug with webkit with how it calculates the proper width of a row when a column has a `colspan` larger than the number of columns in the table (this is how the rxBulkSelectHeader element currently works).

### Firefox
ff_corrected_floating

* Works perfectly!

File: src/elements/Tables/styles/tables.less [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    color: @gray-700;
+    cursor: pointer;
+
+    &:hover {
+      color: saturate(@red-700, 15%);
+      cursor: pointer;
+    }//&:hover
+  }//.fa-times.delete-x
+
+  .expanded-container {
```
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: NOTE TO SELF: move out of `table {}`

File: src/elements/Tables/styles/tables.less [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+thead {
+  th {
+    text-align: left;
+    font-weight: 500;
+    background: @app-table-header-background-color;
+    color: @app-table-header-text-color;
+    border-color: @app-table-header-border-color;
+  }
+}//thead
```
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: With the changes to these styles, we were able to reduce CSS specificity across 10 of the 21 unique CSS selectors for table styling.

File: src/components/rxFloatingHeader/scripts/rxFloatingHeader.js [hunk 4]
```diff
@@ -236,9 +235,16 @@ angular.module('encore.ui.rxFloatingHeader')
...
                             clone.remove();
                         });
+
+                        _.each(trs, function (tr) {
+                            tr.css({ 'top': null });
+                        });
+
+                        _.each(ths, function (th) {
+                            th.css({ 'width': null });
+                        });
```
Comment (CITguy) [inline src/components/rxFloatingHeader/scripts/rxFloatingHeader.js]: Remove added `style` properties on table headers, when not floating.

File: src/components/rxFloatingHeader/rxFloatingHeader.less [hunk 5]
```diff
@@ -12,7 +12,7 @@
 th.filter-header {
   font-weight: normal;
   background: rgba(255, 255, 255, 0.95);
-  border: 1px solid @gray-300;
```
Comment (CITguy) [inline src/components/rxFloatingHeader/rxFloatingHeader.less]: Borders handled by table styling only.

File: src/components/rxBulkSelect/rxBulkSelect.less [hunk 6]
```diff
@@ -84,10 +83,6 @@ thead th[rx-bulk-select-header-check] {
 
 .selected td:not(.status) {
   background: @app-selected-row-background-color;
-  border-color: @green-100;
-  // this 'double' makes it so that the border-color applies to the top border as well
-  // @see http://stackoverflow.com/questions/7942212/css-table-row-border-color-with-border-collapse
-  border-style: double;
```
Comment (CITguy) [inline src/components/rxBulkSelect/rxBulkSelect.less]: Borders handled by table styling only.

File: src/components/rxBulkSelect/rxBulkSelect.less [hunk 7]
```diff
@@ -7,7 +7,6 @@ thead th.bulk-select-header {
   text-align: center;
   color: @rxBulkSelect-header-text-color;
   font-weight: normal;
-  border: 1px solid @app-table-border-color;
```
Comment (CITguy) [inline src/components/rxBulkSelect/rxBulkSelect.less]: Borders handled by table styling only.

File: src/components/rxApp/common.less [hunk 8]
```diff
@@ -75,106 +75,6 @@ a:focus {
...
-  }
-  .expanded-content {
-    background-color: @white;
-    margin-top: 3px;
-    .metadata {
-      padding: 5px 10px;
-    }
-  }
-}
-
```
Comment (CITguy) [inline src/components/rxApp/common.less]: Refactored and moved to `src/elements/Tables/styles/tables.less`

File: src/elements/Tables/styles/tables.less [hunk 9]
```diff
@@ -0,0 +1,114 @@
+// With Regards to Table Borders:
+//
+// The <table> will handle the top and left border
+// while the <td> and <th> will handle the bottom and right border
+table {
+  border-collapse: separate;
+  border-color: @app-table-border-color;
+  border-style: solid;
+  border-width: @app-table-border-width 0 0 @app-table-border-width;
```
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: Top and Left borders only

File: src/elements/Tables/styles/tables.less [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+      padding: 5px 10px;
+    }
+  }//.expanded-content
+}//.expanded-container
+
+td,
+th {
+  border-color: @app-table-border-color;
+  border-style: solid;
+  border-width: 0 @app-table-border-width @app-table-border-width 0;
```
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: Right and Bottom borders only.

File: src/elements/Tables/styles/tables.less (hunk 3 above)
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: * `table` (0001)
  * no change necessary
* `table td, table th` (0002)
  * changed to `td, th` (0001 - **lower specificity**)
* `table thead th` (0003)
  * changed to `thead th` (0002 - **lower specificity**)
* `table thead th.actions` (0013)
  * changed to `td.actions, th.actions` (0011 - **lower specificity**)
* `table tr` (0002)
  * changed to `tr` (0001 - **lower specificity**)
* `table.table-striped` (0011)
  * no change necessary
* `table td.double-chevron-cell` (0012)
  * changed to `td.double-chevron-cell` (0011 - **lower specificity**)
* `table td.double-chevron-cell .double-chevron` (0022)
  * changed to `td.double-chevron-cell .double-chevron` (0021 - **lower specificity**)
* `table td.double-chevron-cell .double-chevron::after` (0023)
  * changed to `td.double-chevron-cell .double-chevron::after` (0022 - **lower specificity**)
* `table td.expanded` (0012)
  * changed to `td.expanded` (0011 - **lower specificity**)
* `table td.expanded .double-chevron::after` (0023)
  * changed to `td.expanded .double-chevron::after` (0022 - **lower specificity**)
* `table .fa-times.delete-x` (0021)
  * no change necessary
* `table .fa-times.delete-x:hover` (0031)
  * no change necessary
* `.expanded-container` (0010)
  * no change necessary
* `.expanded-container > *` (0011)
  * no change necessary
* `.expanded-container table.subtable` (0021)
  * changed to `table.subtable` (0011 - **lower specificity**)
* `.expanded-container table.subtable th:first-child` (0032)
  * no longer needed
* `.expanded-container table.subtable th:last-child` (0032)
  * no longer needed
* `.expanded-container table.subtable th` (0022)
  * changed to `th` (0001 - **lower specificity**)
* `.expanded-container .expanded-content` (0020)
  * no change necessary
* `.expanded-container .expanded-content .metadata` (0030)
  * no change necessary

File: src/components/rxPaginate/docs/rxPaginate.html [hunk 11]
```diff
@@ -70,7 +70,7 @@
         </th>
       </tr>
       <tr>
-        <th class="column-title" style="width:10em;">
```
Comment (CITguy) [inline src/components/rxPaginate/docs/rxPaginate.html]: Explicit `width` styling interferes with rxFloatingHeader functionality.

File: src/elements/Tables/styles/tables.less [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+  color: @app-table-cell-text-color;
+  padding: @app-table-cell-padding;
+
+  &:last-child {
+    border-right-color: @app-table-border-color;
+  }
+
+  &.actions {
+    width: 15px;
```
Comment (ssalinas24) [inline src/elements/Tables/styles/tables.less]: I wonder if we should try to make this a percentage as well. I used to do this with Font-sizes and its usually target/context. Looks like the width of things is the same thing. This site has a quick little tool, http://responsv.com/flexible-math/

File: src/elements/Tables/styles/tables.less [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+    &:hover {
+      color: saturate(@red-700, 15%);
+      cursor: pointer;
+    }//&:hover
+  }//.fa-times.delete-x
+}//table
+
+.expanded-container {
+  padding: 0 3px 20px 3px;
```
Comment (ssalinas24) [inline src/elements/Tables/styles/tables.less]: Maybe we could make these percentages as well.

File: src/elements/Tables/styles/tables.less [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+  border-bottom: 1px solid @app-table-header-border-color;
+
+  & > * {
+    margin-left: 2%;
+    width: 98%;
+  }
+
+  .expanded-content {
+    background-color: @white;
+    margin-top: 3px;
```
Comment (ssalinas24) [inline src/elements/Tables/styles/tables.less]: Lets see if we can do this in percentage as well

File: src/elements/Tables/styles/tables.less (hunk 13 above)
Comment (ssalinas24) [inline src/elements/Tables/styles/tables.less]: nah never mind. lets stay away from percentages on padding at the moment. I do want to use something other than pixel though.

**Comment by ssalinas24** [comment]
Design LGTM

File: src/elements/Tables/styles/tables.less [hunk 15]
```diff
@@ -0,0 +1,114 @@
+// With Regards to Table Borders:
+//
+// The <table> will handle the top and left border
+// while the <td> and <th> will handle the bottom and right border
+table {
+  border-collapse: separate;
```
Comment (halkon) [inline src/elements/Tables/styles/tables.less]: May need to look at this between `separate` and `collapse`

File: src/elements/Tables/styles/tables.less [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+td,
+th {
+  border-color: @app-table-border-color;
+  border-style: solid;
+  border-width: 0 @app-table-border-width @app-table-border-width 0;
+  color: @app-table-cell-text-color;
+  padding: @app-table-cell-padding;
+
+  &:last-child {
+    border-right-color: @app-table-border-color;
```
Comment (halkon) [inline src/elements/Tables/styles/tables.less]: Double check

File: src/elements/Tables/styles/tables.less [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+  }//&.expanded
+}//td
+
+thead {
+  th {
+    text-align: left;
+    font-weight: 500;
+    background: @app-table-header-background-color;
+    color: @app-table-header-text-color;
+    border-color: @app-table-header-border-color;
```
Comment (halkon) [inline src/elements/Tables/styles/tables.less]: Double Check

**Comment by parlarjb** [comment]
Vidyo LGTM

File: src/elements/Tables/styles/tables.less (hunk 15 above)
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: yup, useless property. I'll remove it.

File: src/elements/Tables/styles/tables.less (hunk 16 above)
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: Still necessary to fix last cell border color. `th:last-child` is more specific than `thead th`, so this wins. Will add comment.

File: src/elements/Tables/styles/tables.less (hunk 17 above)
Comment (CITguy) [inline src/elements/Tables/styles/tables.less]: This is purely to set inner border color of header cells. No comment necessary.

**Comment by cathysiller** [comment]
Dev LGTM

**Comment by halkon** [comment]
LGTM

**Pull request merged**